src/settypes/sortedvector.jl

```diff
@@ -66,8 +65,30 @@ function Base.union(v1::SortedVector{T}, v2::SortedVector{T}) where {T}
         result_index += 1
     end
     resize!(result, result_index - 1)
-    return SortedVector{T}(result; sorted=true)
+    return result
+end
+
+function Base.union!(v1::SortedVector{T}, v2::SortedVector{T}) where {T}
+    resize!(v1.data, length(v1) + length(v2))
+    union!(v1, v1, v2)
+    return v1
+end
+
+function Base.union(v1::SortedVector{T}, v2::SortedVector{T}) where {T}
+    result = similar(v1.data, length(v1) + length(v2))
+    v = SortedVector{T}(result; sorted=false)
+    union!(v, v1, v2)
+    return v
 end
 
 Base.iterate(v::SortedVector)             = iterate(v.data)
 Base.iterate(v::SortedVector, i::Integer) = iterate(v.data, i)
+
+function ×(a::SortedVector{T}, b::SortedVector{T}) where {T}
+    prod_data = Tuple{T,T}[]
+    sizehint!(prod_data, length(a) * length(b))
```

Comment:
Does this help? The docs mention

> Suggest that collection `s` reserve capacity for **at least** `n` elements.

We have **at most** `length(a) * length(b)` elements, haven't we? The `SortedVector` docstring mentions that it is without duplicates.

Reply:
In fact we're gonna have exactly `length(a) *  length(b)` elements, since neither `a` nor `b` has duplicates. But I coded it differently in the end.